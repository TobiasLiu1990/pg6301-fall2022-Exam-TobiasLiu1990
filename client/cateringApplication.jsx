import { Link, Route, Routes } from "react-router-dom";
import React from "react";

//List all dishes
//Create dishes if admin staff
//Order dish

export function ShowMenu() {

}

function ListPizzas() {
  return null;
}

function Cart() {
  return null;
}

function AddNewPizzaDish() {
  return null;
}

export function CateringApplication() {
  return (
    <Routes>
      <Route path={"/"} element={<ListPizzas />} />
      <Route path={"/order"} element={<Cart />} />
      <Route path={"/new"} element={<AddNewPizzaDish />} />
      <Route path={"*"} element={<h1>Catering side not found!</h1>} />
    </Routes>
  );
}
