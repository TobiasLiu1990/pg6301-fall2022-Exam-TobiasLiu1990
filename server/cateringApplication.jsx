import { Link, Route, Routes } from "react-router-dom";
import { ListPizzas } from "./listPizzas";
import { AddNewPizza } from "./addNewPizza";
import {fetchJSON} from "./fetchJson";

export function ShowMenu() {
  return (
    <div>
      <h1>Our special pizzas of oddities</h1>
      <ul>
        <li>
          <Link to={"/menu"}>Go to our Menu</Link>
        </li>
        <li>
          <Link to={"/menu/new"}>Add a new pizza dish</Link>
        </li>
      </ul>
    </div>
  );
}

function Cart() {
  return null;
}

export function CateringApplication() {
    const pizzaApi = {
        async listPizzas() {
            return await fetchJSON("/api/menu")
        }
    }

  return (
    <Routes>
      <Route path={"/"} element={<ListPizzas pizzaApi={pizzaApi} />} />
      <Route path={"/new"} element={<AddNewPizza />} />
      <Route path={"/order"} element={<Cart />} />
      <Route path={"*"} element={<h1>Catering side not found!</h1>} />
    </Routes>
  );
}
