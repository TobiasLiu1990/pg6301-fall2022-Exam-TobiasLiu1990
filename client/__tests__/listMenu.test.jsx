import React from "react";
import ReactDOM from "react-dom"
import { act } from "react-dom/test-utils";
import {ListPizzas} from "../cateringApplication";


// const pizzas = [
//   {
//     pizza: "pizza 1",
//     price: 100,
//     ingredients: ["tomato", "cheese"],
//     allergens: ["casein"],
//   },
//   {
//     pizza: "pizza 2",
//     price: 500,
//     ingredients: ["tomato", "cheese", "peanuts"],
//     allergens: ["casein", "nuts"]
//   },
// ];

describe("ListPizzas component", () => {
  it("should show loading screen", () => {
    const element = document.createElement("div");
    ReactDOM.render(<ListPizzas/>, element);

    expect(element.innerHTML).toMatchSnapshot();
  });

  it("should show pizzas", async () => {
    const pizzas = [ {pizza: "pizza 1"}, {pizza: "pizza 2"} ]
    const element = document.createElement("div");

    await act(async () => {
      ReactDOM.render(<ListPizzas/>, element);
    });

    expect(
        Array.from(element.querySelectorAll("h3")).map((e) => e.innerHTML)
    ).toEqual(["pizza 1", "pizza 2"])

    expect(element.innerHTML).toMatchSnapshot();
  });



});