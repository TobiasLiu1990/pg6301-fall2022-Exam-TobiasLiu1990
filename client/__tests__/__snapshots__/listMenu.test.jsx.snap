// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListPizzas component should show loading screen 1`] = `"<div>Loading menu...</div>"`;
